resourcematch-app/renderer/_default.page.server.tsx

```diff
@@ -0,0 +1,46 @@
+export { render }
+// See https://vike.dev/data-fetching
+export const passToClient = ['pageProps', 'urlPathname']
+
+import ReactDOMServer from 'react-dom/server'
+import { PageShell } from './PageShell'
+import { escapeInject, dangerouslySkipEscape } from 'vike/server'
+import logoUrl from './logo.svg'
+import type { PageContextServer } from './types'
+
+async function render(pageContext: PageContextServer) {
+  const { Page, pageProps } = pageContext
+  // This render() hook only supports SSR, see https://vike.dev/render-modes for how to modify render() to support SPA
+  if (!Page) throw new Error('My render() hook expects pageContext.Page to be defined')
+  const pageHtml = ReactDOMServer.renderToString(
+    <PageShell pageContext={pageContext}>
+      <Page {...pageProps} />
+    </PageShell>
+  )
+
+  // See https://vike.dev/head
+  const { documentProps } = pageContext.exports
+  const title = (documentProps && documentProps.title) || 'Vite SSR app'
+  const desc = (documentProps && documentProps.description) || 'App using Vite + Vike'
```

Comment:
Minor: These can be tweaked if we want.

Reply:
Updated - was not 100% sure if 500 characters is the actual limit (determined via chatter in Slack). Figure we could always update again if character limit changes or is incorrect.

Reply:
Ah -- I think we said 500 words which would be 2500 chars based on the "usual" 5 chars/word. I think some of the example Matt had were ~3000 chars, so we can probably use 3500 in any validation, etc.